source/Statistics/Overlap.cpp

```diff
@@ -0,0 +1,145 @@
+/* *****************************************************************************
+
+    trimAl v2.0: a tool for automated alignment trimming in large-scale
+                 phylogenetics analyses.
+
+    readAl v2.0: a tool for automated alignment conversion among different
+                 formats.
+
+    2022-2023
+        Larralde M.             (martin.larralde@embl.de)
+
+    2009-2019
+        Fernandez-Rodriguez V.  (victor.fernandez@bsc.es)
+        Capella-Gutierrez S.    (salvador.capella@bsc.es)
+        Gabaldon, T.            (tgabaldon@crg.es)
+
+    This file is part of trimAl/readAl.
+
+    trimAl/readAl are free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, the last available version.
+
+    trimAl/readAl are distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with trimAl/readAl. If not, see <http://www.gnu.org/licenses/>.
+
+***************************************************************************** */
+
+#include "Statistics/Overlap.h"
+#include "InternalBenchmarker.h"
+#include "Statistics/Manager.h"
+#include "reportsystem.h"
+#include "Alignment/Alignment.h"
+#include "defines.h"
+#include "utils.h"
+
+namespace statistics {
+
+    Overlap::Overlap(Alignment *parentAlignment) {
+        StartTiming("Overlap::Overlap(Alignment *parentAlignment) ");
+       
+        alig = parentAlignment;
+        refCounter = new int(1);
+    }
+
+    Overlap::Overlap(Alignment *parentAlignment, Overlap *mold) {
+        StartTiming("Overlap::Overlap(Alignment *parentAlignment, Overlap *mold) ");
+
+        alig = parentAlignment;
+
+        overlaps = mold->overlaps;
+        refCounter = mold->refCounter;
+        (*refCounter)++;
+    }
+
+    Overlap::~Overlap() {
+        if (refCounter == nullptr || --(*refCounter) < 1) {
+            if (overlaps != nullptr) {
+                for (int i = 0; i < alig->numberOfSequences; i++)
+                    delete[] overlaps[i];
+                delete[] overlaps;
+            }
+            delete refCounter;
+            refCounter = nullptr;
+        }
+    }
+
+    void Overlap::calculateSeqOverlap() {
```

Comment:
Actually I don't think so, because in `Cleaner::calculateSpuriousVector` the overlap computation for a sequence `i` is done for all sequences but itself, whereas in `Alignment::calculateSeqOverlap` the overlap computation counts the sequence itself as part of the overlap statistics.